### PR TITLE
DP-1665: encapsulate movai-service

### PR DESCRIPTION
**Pull request by mariana-dias-alves**
Goal:
- encapsulate python dependencies in debian packaging of movai-service and movai-cli so that it works on focal, jammy and noble

What is done:
- re-work cicd to build 3 packages
- re-work cicd to publish to different ppa's (focal, jammy and noble ppa to be created)
     - https://github.com/MOV-AI/movai-service/actions/runs/12157211677
     - https://github.com/MOV-AI/movai-service-temp/actions/runs/12116587834
          - https://artifacts.nexus.es.mov.ai/#browse/browse:ppa-devel-noble:packages%2Fm%2Fmovai-service%2F0.0.0-9
          - https://artifacts.nexus.es.mov.ai/#browse/browse:ppa-testing-noble:packages%2Fm%2Fmovai-service%2F0.0.0-9
          - https://github.com/MOV-AI/movai-service-temp/releases/tag/0.0.0-9
     - https://github.com/MOV-AI/movai-service-temp/actions/runs/12117706518
          - https://artifacts.nexus.es.mov.ai/#browse/browse:ppa-stable-jammy:packages%2Fm%2Fmovai-cli%2F0.0.0-9

-------------------------

- [x] Make sure you are opening from a **topic/feature/bugfix branch**
- [x] Ensure that the PR title represents the desired changes
- [x] Ensure that the PR description detail the desired changes
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

[^note]:
    Put an `x` into the [ ] to show you have filled the information.
    The template comes from https://github.com/MOV-AI/.github/blob/master/.github/pull_request_template.md
    You can override it by creating .github/pull_request_template.md  in your own repository

### CODE REVIEW

File: .github/workflows/service-py-deb-workflow.yml [hunk 1]
```diff
@@ -330,64 +330,39 @@ jobs:
...
       uses: actions/checkout@v4
       with:
+        ref: ${{ needs.Raise.outputs.commit_id }}
         submodules: recursive
-
-    - name: Download a single artifact
-      uses: actions/download-artifact@v4
+    
+    - name: Login to registry.cloud.mov.ai Registry
+      uses: docker/login-action@v2
```
Comment (AlexFernandes-MOVAI) [inline .github/workflows/service-py-deb-workflow.yml]: use v3
```suggestion
      uses: docker/login-action@v3
```

File: .github/workflows/service-py-deb-workflow.yml [hunk 2]
```diff
@@ -509,7 +477,7 @@ jobs:
       - name: Download deb artifacts
         uses: actions/download-artifact@v4
         with:
-          pattern: deb_artifacts_*
+          pattern: deb_artifacts_${{ matrix.distro }}
           path: artifacts
```
Comment (AlexFernandes-MOVAI) [inline .github/workflows/service-py-deb-workflow.yml:387]: use a specific folder per distribution not to overwrite binaries

File: .github/workflows/service-py-deb-workflow.yml [hunk 3]
```diff
@@ -693,9 +661,13 @@ jobs:
...
         echo "Please check report here: https://sonarcloud.io/project/overview?id=${{ github.repository_owner }}_${{ github.event.repository.name }}"
 
-  Publish:
+  PublishDebian:
     needs: [Raise, Debian-Pack, Rpm-Pack, Debian-Test, SonarCloud]
     if: ${{ inputs.deploy == 'true' }}
+    strategy:
+      matrix:
+        distro: ${{ fromJSON(inputs.distributions) }}
+      fail-fast: false
```
Comment (AlexFernandes-MOVAI) [inline .github/workflows/service-py-deb-workflow.yml]: fail-fast is for debug only

File: .github/workflows/service-py-deb-workflow.yml [hunk 4]
```diff
@@ -791,6 +749,28 @@ jobs:
               fi
             done
 
+  PublishRpm:
```
Comment (AlexFernandes-MOVAI) [inline .github/workflows/service-py-deb-workflow.yml]: remove or disable all Rpm (we won't maintain it anymore)

File: .github/workflows/service-py-deb-workflow.yml [hunk 5]
```diff
@@ -875,67 +984,28 @@ jobs:
...
-        path: artifacts/*.rpm
-        retention-days: 5
-
-    - name: Publish package to prod PyPI
-      uses: pypa/gh-action-pypi-publish@v1.11.0
-      with:
-        user: ${{ secrets.nexus_publisher_user }}
-        password: ${{ secrets.nexus_publisher_password }}
-        repository-url: https://artifacts.cloud.mov.ai/repository/${{ inputs.prod_publish_pypi }}/
+        path: artifacts/
```
Comment (AlexFernandes-MOVAI) [inline .github/workflows/service-py-deb-workflow.yml:773]: use different paths per distribution

**Review by AlexFernandes-MOVAI — CHANGES_REQUESTED** [review]

**Comment by duartecoelhomovai** [comment]
My only comment is to take this pipeline to v2.2

File: .github/workflows/service-py-deb-workflow.yml [hunk 6]
```diff
@@ -330,64 +330,39 @@ jobs:
     if: ${{ inputs.release == 'false' }}
```
Comment (AlexFernandes-MOVAI) [inline .github/workflows/service-py-deb-workflow.yml:307]: Debian package does not contain the correct version, see the step where bumpversion is called and try something like this if you wanna keep calculating the version 
```
	bump2version --no-tag --allow-dirty --new-version $(PACKAGE_VERSION)
```

or just `bump2version build --no-tag --allow-dirty --commit-args no-verify` to bump the build id only

**Review by AlexFernandes-MOVAI — CHANGES_REQUESTED** [review]

**Pull request closed without merging**